Detectors/TRD/base/include/TRDBase/CalVdriftExB.h

```diff
@@ -0,0 +1,56 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_TRD_VDRIFTEXB_H
+#define O2_TRD_VDRIFTEXB_H
+
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+//  TRD calibration class for parameters which are saved frequently(/run)    //
+//  2019 - Ported from various bits of AliRoot (SHTM)                        //
+//  Most things were stored in AliTRDcalROC,AliTRDcalPad, AliTRDcalDet       //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "TRDBase/SimParam.h"
+#include "DataFormatsTRD/Constants.h"
+#include "TRDBase/Geometry.h"
+//
+class Geometry;
+
+namespace o2
+{
+namespace trd
+{
+class CalVdriftExB
+{
+ public:
+  CalVdriftExB() = default;
+  ~CalVdriftExB() = default;
+  //
+  float getVDrift(int p, int c, int s) const { return mVDrift[o2::trd::Geometry::getDetector(p, c, s)]; };
```

Comment:
Why would we need all these setters and getters at all for values which are only stored per chamber? Whereever the calibration parameters are queried they are queried for a full chamber, are they not?

Reply:
i think in the line just below there is a getter to get a value by i_det, and this is an alternative option to get it by layer, stack, sector number? same for setters

Reply:
either way the p, c, and s should at least be expanded. The person that originally wrote this with the p, c, and s should be keel hauled ;-)

Reply:
Agree with @bazinski 
But on topic: all we need are 4 arrays with one value per chamber. This could be a simple `struct` which then can go into `DataFormatsTRD`. No need for any dependency on simulation parameters or geometry. No need for any setters or getters. If one would need to go from p, c, s (whatever these values mean) to detector number then this can be done at the place where these values are queried. Because at those places one also knows about the TRD geometry (for example in the TRD digitizer)

Reply:
i concur with @martenole about removing the p,c,s version of getVDrift and thereby removing dependencies.